modin/pandas/base.py

```diff
@@ -2082,14 +2082,18 @@ def _reindex(
         """
         new_query_compiler = None
         if index is not None:
-            if not isinstance(index, pandas.Index):
-                index = self._copy_index_metadata(
-                    source=self.index, destination=self._ensure_index(index, axis=0)
-                )
-            if not index.equals(self.index):
-                new_query_compiler = self._query_compiler.reindex(
-                    axis=0, labels=index, **kwargs
-                )
+            # TODO(REFACTOR): see how/whether to detect modin.pandas index in a better
+            # way. Right now in upsteram modin we always try to convert the index to a
+            # pandas.Index. That is not desirable for large indexes that we store in the
+            # service.
+            # Use duck typing to detect modin.pandas indexes, which are rpyc netrefs.
+            # pandas.Index and modin.pandas.Index and not much else has an attribute
+            # called "_summary".
+            if hasattr(index, "_summary"):
```

Comment:
So we can't use `isinstance` because the index is a netref, is that correct?

Reply:
that's right. There could be more sophisticated ways of presenting the index to the user so they can know what's going on (we should probably share the index class definition with them and move the sensitive details to private methods) but this works for now.